delivery_roulier/models/stock_picking.py

```diff
@@ -0,0 +1,249 @@
+        return:
+            dict
+        """
+        address = {}
+        extract_fields = [
+            'company', 'name', 'zip', 'city', 'phone', 'mobile',
+            'email', 'street2']
+        for elm in extract_fields:
+            if elm in partner:
+                # because a value can't be None in odoo's ORM
+                # you don't want to mix (bool) False and None
+                if partner._fields[elm].type != fields.Boolean.type:
+                    if partner[elm]:
+                        address[elm] = partner[elm]
+                    # else:
+                    # it's a None: nothing to do
+                else:  # it's a boolean: keep the value
+                    address[elm] = partner[elm]
+        if not address.get('company', False) and partner.parent_id.is_company:
+            address['company'] = partner.parent_id.name
+        # Roulier needs street1 (mandatory) not street
+        address['street1'] = partner.street
+        # Codet ISO 3166-1-alpha-2 (2 letters code)
+        address['country'] = partner.country_id.code
+
+        for tel in ['mobile', 'phone']:
+            if address.get(tel):
+                address[tel] = address[tel].replace(u'\u00A0', '').\
+                    replace(' ', '')
+
+        address['phone'] = address.get('mobile', address.get('phone'))
+
+        return address
+
+    def _roulier_get_from_address(self, package=None):
+        sender = self._get_sender(package=package)
+        return self._convert_address(sender)
+
+    def _roulier_get_to_address(self, package=None):
+        receiver = self._get_receiver(package=package)
+        return self._convert_address(receiver)
+
+    def _roulier_get_service(self, account, package=None):
+        """Return a basic dict.
+
+        The carrier implementation may add stuff
+        like agency or options.
+
+        return:
+            dict
+        """
+        shipping_date = self._get_shipping_date(package)
+
+        service = {
+            'product': self.carrier_code,
+            'shippingDate': shipping_date,
+            'labelFormat': self._get_label_format(account)
+        }
+        return service
+
+    def open_website_url(self):
+        """Open tracking page.
+
+        More than 1 tracking number: display a list of packages
+        Else open directly the tracking page
+        """
+        self.ensure_one()
+        if not self._is_roulier():
+            return super().open_website_url()
+
+        packages = self._get_packages_from_picking()
+        if len(packages) == 0:
+            raise UserError(_('No packages found for this picking'))
+        elif len(packages) == 1:
+            return packages.open_website_url()  # shortpath
+
+        # display a list of pickings
+        action = self.env.ref('stock.action_package_view').read()[0]
+        action['res_id'] = packages.ids
+        action['domain'] = "[('id', 'in', [%s])]" % (
+            ",".join(map(str, packages.ids))
+        )
```

Comment:
why don't you just pass the domain as python list?

Reply:
no reason, I change it